src/irmin-pack/unix/io_errors.ml

```diff
@@ -24,54 +24,75 @@ module type S = sig
 
   type t = [ Base.t | `Io_misc of Io.misc_error ]
```

Comment:
`type t = [ Base.t | `Io_misc of Io.misc_error ] [@@deriving irmin` is the typical way to go